conbench/tests/app/test_compare.py

```diff
@@ -39,6 +39,41 @@ def test_flash_messages(self, client):
         assert "cannot perform comparison:" in response.text
 
 
+class TestCompareBenchmarks(_asserts.AppEndpointTest):
+    url = "/compare/benchmarks/{}/"
+    title = "Compare Benchmark Results"
+    redirect_on_unknown = False
+
+    def _create(self, client):
+        benchmark_result_id = self.create_benchmark(client)
+        return f"{benchmark_result_id}...{benchmark_result_id}"
+
+    def test_redirects(self, client):
+        self.authenticate(client)
+
+        response = client.get(
+            "/compare/benchmarks/unknown...unknown2/", follow_redirects=True
+        )
+        assert "/compare/benchmark-results/unknown...unknown2/" in response.request.url
+        assert any(
+            [
+                "/compare/benchmarks/unknown...unknown2/" == res.request.path
+                for res in response.history
+            ]
+        )
+        assert "cannot perform comparison:" in response.text, response.text
+
+        response = client.get("/compare/benchmarks/foo...bar/", follow_redirects=True)
+        assert "/compare/benchmark-results/foo...bar/" in response.request.url
```

Comment:
I mean, it's OK :)

But I think it's wort pointing out that this tests an implementation detail (exhaustively, even!) that we don't care about too much.

We care about that the route works, now really how it's doing so.

My preference would be to not encode implementation details like this in a test.

We already have a number of tests that probe for implementation details, and this gets in the way when changing things, need to fight this trend a bit.

Just for comparison and towards a better future: a simpler, a more focused way to test here would be 

```python
def test_legacy_route():
    resp = requests.get('http://testhost:1337/compare/benchmarks/foo...bar')
    assert res.status_code == 200, ...
    assert "cannot perform comparison:" in resp.text
```

That would test the interface from the user's point of view. It makes use of the fact that most HTTP clients follow redirects by default (requests.get() does).